csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.shared.cs

```diff
@@ -275,6 +275,19 @@ public void AppendExecutionProvider_DML(int deviceId = 0)
 #endif
         }
 
+        /// <summary>
+        /// Use only if you have the onnxruntime package specific to this Execution Provider.
+        /// </summary>
+        /// <param name="dmlDevice">A IDMLDevice reference</param>
+        /// <param name="commandQueue">A ID3D12CommandQueue reference</param>
+        public void AppendExecutionProvider_DML1(IntPtr dmlDevice, IntPtr commandQueue)
```

Comment:
Generally, we do not expose native IntPtr handles via public API. The preferred way is to expose an object of DmlDevice that has a handle.
That object is likely to be IDisposable.
Then that object can expose an `internal` property Handle that can be used for NativeApi.
The same probably goes for commandQueue.